src/TypeCatalogGen/TypeCatalogGen.cs

```diff
@@ -242,16 +245,27 @@ private static string GetAssemblyStrongName(MetadataReader metadataReader)
                 case AssemblyHashAlgorithm.Sha512:
                     hashImpl = SHA512.Create();
                     break;
+                case AssemblyHashAlgorithm.None:
+                    Console.WriteLine($@"No HashAlgorithm skipping: {asmName}");
+                    break;
                 default:
                     throw new NotSupportedException();
             }
 
-            byte[] publicKeyHash = hashImpl.ComputeHash(publickey);
-            byte[] publicKeyTokenBytes = new byte[8];
-            // Note that, the low 8 bytes of the hash of public key in reverse order is the public key tokens.
-            for (int i = 1; i <= 8; i++)
+            byte[] publicKeyTokenBytes;
+            if (hashImpl != null)
+            {
+                byte[] publicKeyHash = hashImpl.ComputeHash(publickey);
+                publicKeyTokenBytes = new byte[8];
```

Comment:
This should be Span<byte> and `stackalloc`.

Reply:
This is existing code, we can refactor this in a separate PR.

Reply:
It's a small change, I'll do it in this PR since I have to wait for PowerShell-Native update anyways